### PR TITLE
libyaml &yaml-py: update to newer upstream version

### PR DESCRIPTION
**Pull request by kamischi**
yaml-py: update to 3.13 and adding py37 variant
libyaml: update to 0.2.1

### CODE REVIEW

**Comment by nieder** [comment]
The libyaml part looks fine. Will yaml-py need a versioned dep on libyaml? I haven't tested building that.

**Comment by kamischi** [comment]
I have checked, whether yaml-py-3.13 can be built and installed with libyaml-0.1.6. There are no obvious problems. Unless something shows up, no versioned dep on libyaml is required, unless your policy is different.

Mischi

**Comment by nieder** [comment]
Add tests to yaml-py?
````
InfoTest: <<
	TestScript: python%type_raw[python] setup.py test || exit 2
<<
````

**Comment by kamischi** [comment]
Test fail is beyond my python capabilities:

/sw/bin/python3.6 setup.py test || exit 2
running test
running build_py
running build_ext
skipping 'ext/_yaml.c' Cython extension (up-to-date)
.....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................FF..F.....F.........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
===========================================================================
test_emitter_styles_ext(tests/data/spec-07-09.data, tests/data/spec-07-09.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 240, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo'),
 ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo %YAML 1.1'))
---------------------------------------------------------------------------
tests/data/spec-07-09.data:
---
foo
...
# Repeated end marker.
...
---
bar
# No end marker.
---
baz
...
---------------------------------------------------------------------------
tests/data/spec-07-09.canonical:
%YAML 1.1
---
!!str "foo"
%YAML 1.1
---
!!str "bar"
%YAML 1.1
---
!!str "baz"
===========================================================================
test_emitter_styles_ext(tests/data/spec-07-10.data, tests/data/spec-07-10.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 240, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='Root flow scalar'),
 ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='Root flow scalar %YAML 1.1'))
---------------------------------------------------------------------------
tests/data/spec-07-10.data:
"Root flow
 scalar"
--- !!str >
 Root block
 scalar
---
# Root collection:
foo : bar
... # Is optional.
---
# Explicit document may be empty.
---------------------------------------------------------------------------
tests/data/spec-07-10.canonical:
%YAML 1.1
---
!!str "Root flow scalar"
%YAML 1.1
---
!!str "Root block scalar\n"
%YAML 1.1
---
!!map {
  ? !!str "foo"
  : !!str "bar"
}
---
#!!str ""
!!null ""
===========================================================================
test_emitter_styles_ext(tests/data/spec-07-13.data, tests/data/spec-07-13.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 240, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='!foo', implicit=(False, False), value='No directives'),
 ScalarEvent(anchor=None, tag='!foo', implicit=(False, False), value='No directives %TAG ! %21foo'))
---------------------------------------------------------------------------
tests/data/spec-07-13.data:
! "First document"
---
!foo "No directives"
%TAG ! !foo
---
!bar "With directives"
%YAML 1.1
---
!baz "Reset settings"
---------------------------------------------------------------------------
tests/data/spec-07-13.canonical:
%YAML 1.1
---
!!str "First document"
---
!<!foo> "No directives"
---
!<!foobar> "With directives"
---
!<!baz> "Reset settings"
===========================================================================
test_emitter_styles_ext(tests/data/spec-08-08.data, tests/data/spec-08-08.canonical): FAILURE
Traceback (most recent call last):
  File "tests/lib3/test_appliance.py", line 59, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib3/test_yaml_ext.py", line 240, in wrapper
    function(*args, **kwds)
  File "tests/lib3/test_emitter.py", line 62, in test_emitter_styles
    _compare_events(events, new_events)
  File "tests/lib3/test_emitter.py", line 15, in _compare_events
    assert event1.value == event2.value, (event1, event2)
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
(ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo bar'),
 ScalarEvent(anchor=None, tag='tag:yaml.org,2002:str', implicit=(False, False), value='foo bar %YAML 1.1'))
---------------------------------------------------------------------------
tests/data/spec-08-08.data:
---
foo:
 "bar
 baz"
---
"foo
 bar"
---
foo
 bar
--- |
 foo
...
---------------------------------------------------------------------------
tests/data/spec-08-08.canonical:
%YAML 1.1
---
!!map {
  ? !!str "foo"
  : !!str "bar baz"
}
%YAML 1.1
---
!!str "foo bar"
%YAML 1.1
---
!!str "foo bar"
%YAML 1.1
---
!!str "foo\n"
===========================================================================
TESTS: 2585
FAILURES: 4
error: Tests failed
### execution of /sw/bin/python3.6 failed, exit code 2
### execution of /tmp/fink.MEKe2 failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/sw/bin/dpkg-lockwait -r fink-buildlock-yaml-py36-3.13-1

**Comment by dmacks** [comment]
Was reported upstream (linux flagged it too) and apparently fixed by https://github.com/yaml/pyyaml/pull/122

**Comment by nieder** [comment]
Ah. I tested with the old libyaml-0.1.6, where it all passed.

**Comment by kamischi** [comment]
To the best of my knowledge, the patch to libyaml-0.2.1 fixes the tests in yaml-py. I hope i got that patch right.

As this is more than trivial changes to yaml-py, I recommend to wait for Kurt Schwehr's OK.

**Comment by nieder** [comment]
With this fix, yaml-py should version its (Build)Depends on libyaml(-shlibs) to 0.2.1-2 to make sure it uses the version with the fix.

**Comment by kamischi** [comment]
That should do it.

**Comment by nieder** [comment]
Ping @schwehr for the new changes to libyaml.

File: 10.9-libcxx/stable/main/finkinfo/languages/libyaml.info [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
+PatchFile-MD5: 2fdfd0f847760e0c46a6104dfedc4fd9
+
 PatchScript: <<
-	# Patch configure to not link like Puma on Yosemite
-	perl -pi.bak -e 's|10\.\[012\]\*|10.[012][,.]*|g' configure
+  # Patch configure to not link like Puma on Yosemite
+  perl -pi.bak -e 's|10\.\[012\]\*|10.[012][,.]*|g' configure
+
+  # Patch from upstream to fix test with yaml-py
+  # can possibly be removed with next version
```
Comment (schwehr) [inline 10.9-libcxx/stable/main/finkinfo/languages/libyaml.info]: Change this to `# can possibly be removed post version 0.2.1`

File: 10.9-libcxx/stable/main/finkinfo/languages/libyaml.info [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
+PatchFile: %n.patch
+PatchFile-MD5: 2fdfd0f847760e0c46a6104dfedc4fd9
+
 PatchScript: <<
-	# Patch configure to not link like Puma on Yosemite
-	perl -pi.bak -e 's|10\.\[012\]\*|10.[012][,.]*|g' configure
+  # Patch configure to not link like Puma on Yosemite
+  perl -pi.bak -e 's|10\.\[012\]\*|10.[012][,.]*|g' configure
+
+  # Patch from upstream to fix test with yaml-py
```
Comment (schwehr) [inline 10.9-libcxx/stable/main/finkinfo/languages/libyaml.info]: Where did the version come from (give a URL)?  I think just replying to this comment is enough documentation.

Comment (kamischi) [reply 10.9-libcxx/stable/main/finkinfo/languages/libyaml.info]: I deduced it from the report  yaml/libyaml#122, mentioned above. Corresponding comment added to libyaml.info.

**Pull request merged**